ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java

```diff
@@ -284,6 +294,19 @@ public void processPriorCommands() {
           .getKsqlEngine()
           .getPersistentQueries();
 
+      final Set<String> stateStoreNames =
```

Comment:
nit: this feels a little out of place at the top level of this method, and also leaks some internal details about the underlying stream processing runtime into this class. Could we break it into it's own interface like this that we pass into this class and call here?

```
class PersistentQueryCleanup {
    void cleanupLeakedQueries(List<PersistentQueryMetadata> activeQueryIds);
}
```

Reply:
This interface would get initialized from `ksqlServerRestApplication`, right? So we'd get the `stateDir` and `serviceContext` from there? What's the benefit of doing this as an interface instead of a class, and if we do an interface is the idea that it's implemented in `ksqlServerRestApplication` or `commandRunner`?

Reply:
We'd implement the interface from `KsqlServerRestApplication`, which is the benefit. Ideally we should avoid writing code in this class that assumes the underlying runtime is KafkaStreams.

Reply:
I'm not sure I follow - what's the benefit of doing `PersistentQueryCleanup` as an interface in `KsqlServerRestApplication` instead of a stand-alone class?

Reply:
I think it's a good goal to try to keep the kafka-streams specific code abstracted away from the rest of the server/engine. It keeps things decoupled which makes the code easier to understand and reason about. Putting an interface between the command runner and the streams code that does clean up makes it easier to potentially try out different runtimes in the future. We'll probably never do this (might be a fun hack day project 😄) but it's a good principle to try to structure the code this way.

Reply:
I made a pass at doing an interface - I can't tell if it's what you want. I get that we want to keep streams code abstracted, but it's odd to me that we'd want to implement the `PersistentQueryCleanup` inside of the rest app if we're trying to keep it separate, it makes more sense to me to create a standalone class for `PersistenteQueryCleanup` that the rest app instantiates and then passes to the command runner, who then calls a method in the standalone class, rather than doing a callback to a method inside of the rest app (which is what called the command runner in the first place).

The other option for interface would be to have `KsqlRestApplication` as a whole implement `PersistentQueryCleanup` but this seemed like it would make testing harder since there wouldn't be a simple constructor for a `PersistentQueryCleanup` object. LMK your thoughts

Reply:
> but it's odd to me that we'd want to implement the PersistentQueryCleanup inside of the rest app if we're trying to keep it separate, it makes more sense to me to create a standalone class for PersistenteQueryCleanup that the rest app instantiates and then passes to the command runner

Either way is fine by me - putting it in a standalone class is probably cleaner. It would be good to have an interface the standalone class implements to keep the implementation completely separate.